src/Symfony/Component/Messenger/Middleware/RouterContextMiddleware.php

```diff
@@ -49,10 +48,20 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             return $stack->next()->handle($envelope, $stack);
         }
 
-        $currentContext = $this->router->getContext();
+        $context = $this->router->getContext();
+        $currentContext = [
+            $context->getBaseUrl(),
+            $context->getMethod(),
+            $context->getHost(),
+            $context->getScheme(),
+            $context->getHttpPort(),
+            $context->getHttpsPort(),
+            $context->getPathInfo(),
+            $context->getQueryString(),
+        ];
 
         /* @var RouterContextStamp $contextStamp */
-        $this->router->setContext(new RequestContext(
+        $context->__construct(
```

Comment:
This is super dirty...

Reply:
maybe we should at least add a new method on the `RequestContext` class?

Reply:
Replaced by calls to setters.

Both classes are not in the same package. Adding a new method would require to bump minimal requirements.
I'm not sure it worth it.